### PR TITLE
Issue 17850 reserved vars

### PR DESCRIPTION
**Pull request by dsilvam**
Reserved vars for content types. Applies only for non-system types.

Integration tests were added.

### CODE REVIEW

**Comment by dsilvam** [comment]
This is breaking the startup of dotcms, probably because of the reserved `Host` type variable. Host is not created anymore on startup

**Review by jdotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java [hunk 1]
```diff
@@ -243,7 +294,7 @@ public String suggestVelocityVar(final String tryVar) throws DotDataException {
     for (int i = 1; i < 100000; i++) {
       dc.setSQL(this.contentTypeSql.SELECT_COUNT_VAR);
       dc.addParam(varName.toLowerCase());
-      if (dc.getInt("test") == 0) {
+      if (dc.getInt("test") == 0 && !reservedContentTypeVars.contains(varName.toLowerCase())) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java]: I would do a NPE checking

File: dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java [hunk 2]
```diff
@@ -315,7 +366,10 @@ private ContentType dbSaveUpdate(final ContentType saveType) throws DotDataExcep
     }
 
     ContentType retType = builder.build();
-
+    if(reservedContentTypeVars.contains(retType.variable().toLowerCase())){
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java]: I would do a NPE checking

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 3]
```diff
@@ -631,40 +631,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
...
-
-            runNoLicense(() -> {
-                final GraphQLSchema schema = APILocator.getGraphqlAPI().getSchema();
-                assertNull(schema.getType(testCase._1));
-            });
-        } finally {
-            if(customType!=null) {
-                APILocator.getContentTypeAPI(APILocator.systemUser()).delete(customType);
+        List<Tuple2<String, BaseContentType>> eeTypesList = eeTypes.stream().map((type)->
+                        new Tuple2<>("my"+type.variable(), type.baseType())).collect(Collectors.toList());
```
Comment (jdotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:643]: this is usually done by Tuple.of, but it is ok I guess

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 4]
```diff
@@ -644,40 +644,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
...
-    @DataProvider
-    public static List<Object> dataProviderEEContentTypes() throws Exception {
-        // data provider needs stuff to get initialized because of API access
-        IntegrationTestInitService.getInstance().init();
+    @Test
+    public void testGetSchema_GivenNoEELicense_EnterpriseTypesShouldNotBeAvailableInSchema() throws Exception{
 
         // filter only Enterprise content types
-        final List<ContentType> eeTypes = APILocator
+        List<ContentType> eeTypes = APILocator
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:651]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 5]
```diff
@@ -644,40 +644,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
 
     }
 
-    @DataProvider
-    public static List<Object> dataProviderEEContentTypes() throws Exception {
-        // data provider needs stuff to get initialized because of API access
-        IntegrationTestInitService.getInstance().init();
+    @Test
+    public void testGetSchema_GivenNoEELicense_EnterpriseTypesShouldNotBeAvailableInSchema() throws Exception{
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:648]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 6]
```diff
@@ -644,40 +644,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
...
-                    testCase._2);
-
-            runNoLicense(() -> {
-                final GraphQLSchema schema = APILocator.getGraphqlAPI().getSchema();
-                assertNull(schema.getType(testCase._1));
-            });
-        } finally {
-            if(customType!=null) {
-                APILocator.getContentTypeAPI(APILocator.systemUser()).delete(customType);
+        List<Tuple2<String, BaseContentType>> eeTypesList = eeTypes.stream().map((type)->
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:655]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java [hunk 7]
```diff
@@ -644,40 +644,33 @@ public void testGetSchema_BothSidedRelationshipFields(final RELATIONSHIP_CARDINA
...
+                        new Tuple2<>("my"+type.variable(), type.baseType())).collect(Collectors.toList());
+
+        for (Tuple2<String, BaseContentType> testCase : eeTypesList) {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:658]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:651]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:648]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:655]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:658]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:651]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:648]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:655]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:658]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:651]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:648]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:655]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:658]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:651]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:648]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:655]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:658]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:651]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:648]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:655]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:658]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:651]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:648]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:655]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:658]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:651]: Codacy Issue found: [Local variable 'eeTypes' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:648]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:655]: Codacy Issue found: [Local variable 'eeTypesList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

File: dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/graphql/business/GraphqlAPITest.java:658]: Codacy Issue found: [Local variable 'testCase' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4984974)

**Pull request merged**